extension/encoding/awslogsencodingextension/config.go

```diff
@@ -79,5 +83,10 @@ func (cfg *Config) Validate() error {
 	if len(errs) > 0 {
 		return errors.Join(errs...)
 	}
-	return nil
+	for _, format := range supportedLogFormats {
+		if cfg.Format == format {
+			return nil
+		}
+	}
+	return fmt.Errorf("unsupported format %q, expected one of %q", cfg.Format, supportedLogFormats)
```

Comment:
This part is an extra, covered in the `switch` above. I will review the rest once the PR gets marked for review :)